protocol/messenger.go

```diff
@@ -2485,49 +2488,66 @@ func (m *Messenger) DeleteAccount(address types.Address) error {
 	return m.syncWallets(accs, m.dispatchMessage)
 }
 
-// syncWallets syncs all wallets with paired devices
-func (m *Messenger) syncWallets(accs []*accounts.Account, rawMessageHandler RawMessageHandler) error {
-	if !m.hasPairedDevices() {
-		return nil
-	}
-
-	ctx, cancel := context.WithTimeout(context.Background(), 5*time.Second)
-	defer cancel()
-
-	clock, chat := m.getLastClockWithRelatedChat()
-
+func (m *Messenger) prepareSyncWalletAccountsMessage(accs []*accounts.Account, clock uint64) *protobuf.SyncWalletAccounts {
 	accountMessages := make([]*protobuf.SyncWalletAccount, 0)
 	for _, acc := range accs {
-		if acc.Type != accounts.AccountTypeWatch &&
-			acc.Type != accounts.AccountTypeGenerated {
+		if acc.Chat {
 			continue
 		}
+
+		// Once mobile app supports seed phrase and private key imported accounts we should remove the following `if` block
+		if isMobileApp {
```

Comment:
Could you elaborate why we need this?

Reply:
There are 2 contexts where `isMobileApp` is used:
- when the app is going to dispatch a wallet accounts message and 
- when the app is going to handle a received wallet accounts message (but not sure how mobile app sets the account's type for all possible accounts' types and that's why I added this `if` here to keep the code as it was in case of mobile app).

My intention was to check `isMobileApp` to avoid any potential issues in mobile app and keep the code as it was before cause as far as I know mobile app is handling differently some account types (was discussed [here](https://github.com/status-im/status-go/pull/3245#discussion_r1122536029)) and not sure that it supports derivation from imported accounts and that it has import private key feature.

Reply:
By checking `isMobileApp` the following sync is done:
- `mobile` to `mobile`, mobile app sends default wallet acc, generated acc from the default wallet account and wo accounts, other mobile app handles them
- `mobile` to `desktop`, mobile app sends default wallet acc, generated acc from the default wallet account and wo accounts, desktop app handles them cause it's able to handle any account type
- `desktop` to `mobile`, desktop app sends all accounts, mobile app handles only default wallet acc, generated acc from the default wallet account and wo accounts
- `desktop` to `desktop`, one desktop app sends all accounts, another desktop app handles them all cause it's able to handle any account type

Reply:
Also you can see that `LatestDerivedPath` is set and used only in case of mobile app, the desktop app doesn't use it. Once the mobile app aligns with the desktop app in terms of wallet accounts we will need just to remove those `if`s (that's why I added comments next to them).

Reply:
I would get rid of `isMobile`, we don't currently use wallet at the moment, and it's better address any potential issue now, instead of later, when we forgot all about the code ;) 
I think it should just work fine, since wallet is not currently used, so as long as it's not crashing etc, we should be good, what do you think?

Reply:
@cammellos thinking just now, what about this part [here](https://github.com/status-im/status-go/blob/76813cef3a992e75452d6b5fc2fccd0e53280e1b/protocol/messenger_handler.go#L2864-L2886)?
`LatestDerivedPath` from `settings` is used only in mobile app, the desktop app doesn't use it.

Reply:
May I remove that and expect that mobile app will align with the desktop app soon?

Reply:
Yes, I think we should do that, since wallet in mobile is not used, we'll have to align with desktop before it's usable again

Reply:
`isMobileApp` is completely removed in [this commit](https://github.com/status-im/status-go/pull/3416/commits/6d0b1b04d050e8f7fae1426397153257fb91486f)